### PR TITLE
[runtime] Enable compact seq point by default

### PR DESCRIPTION
**Pull request by esdrubal**
Enable compact seq points by default so that the stack trace have IL
offsets instead os native ones.

This changes will allow any stacktrace generated from now on to be
symbolicated using mono-symbolicate.

### CODE REVIEW

File: man/mono.1 [hunk 1]
```diff
@@ -1652,11 +1652,11 @@ Automatically generates sequence points where the
 IL stack is empty.  These are places where the debugger can set a
 breakpoint.
 .TP
-\fBgen-compact-seq-points\fR
```
Comment (esdrubal) [inline man/mono.1]: yes, it was shipped in a release. It was used in order to mono-symbolicate to work. We can keep it but it won't do anything. @kumpera what do you think, should we keep it?

Comment (akoeplinger) [inline man/mono.1]: It's used in monodroid at least: https://github.com/xamarin/xamarin-android/blob/14260f43fc47abc9aee0ff5a0ca10a10ac2e3384/src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs#L324 (and who knows what other code relies on it).

We should just make this a no-op now that it's the default.

Comment (akoeplinger) [inline man/mono.1]: And now that I look at it, the same likely applies to `gen-seq-points-file` as well which was removed in https://github.com/mono/mono/commit/eb8cc5ca57c7fd03ca0026c9c0bcef9734c78776. 

It's used in monodroid here: https://github.com/xamarin/xamarin-android/blob/14260f43fc47abc9aee0ff5a0ca10a10ac2e3384/src/Xamarin.Android.Build.Tasks/Tasks/Aot.cs#L329

Comment (marek-safar) [inline man/mono.1]: yeah, we cannot simple drop it without notice especially when it was documented. Put it back and make it do nothing with a warning of deprecation

Comment (esdrubal) [inline man/mono.1]: thanks for the input. In the new commits both `MONO_DEBUG=gen-compact-seq-points` and `--aot=gen-seq-points-file=<dir>` are still available but with deprecated warning.

**Pull request merged**